src/GetMarkdownMetadataCommand.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.PlatyPS
+{
+    /// <summary>
+    /// Get-MarkdownMetadata reads the YAML header from a markdown file and represents it as a Dictionary object
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "MarkdownMetadata", HelpUri = "", DefaultParameterSetName = "FromPath")]
+    [OutputType(typeof(Dictionary<object, object>))]
+    public sealed class GetMarkdownMetadataCommand : PSCmdlet
+    {
+        #region cmdlet parameters
+
+        /// <summary>
+        /// An array of paths to get the markdown metadata from.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "FromPath",
+            ValueFromPipelineByPropertyName = true,
+            Position = 1)]
+        [SupportsWildcards]
+        public string[]? Path { get; set; }
```

Comment:
> Assigning null to a non-nullable reference type -- won't this result in a compilation error?

We should assign `null!`, not `null`. This works well:
`public string[] Path { get; set; } = null!;`

Reply:
Ah, yes, the null-forgive operator. But it looks weird though -- declaring that the `null` expression isn't null 😄

Reply:
Initialization with Array.Empty() confuses me more :-)

---

If we declare an _public_ API non-nullable we _must_ check it is not null in any case. 
Here we do the check with attribute. So the pattern is good and we can use it everywhere in PowerShell repository.

Reply:
> So the pattern is good and we can use it everywhere in PowerShell repository.

Hold on, I still think `null!` is counter intuitive :)